src/app/context/AuthProvider.tsx

```diff
@@ -0,0 +1,9 @@
+'use client'
+
+import { SessionProvider } from 'next-auth/react'
+
+import { ReactNode } from 'react'
+
+export const AuthProvider = ({ children }: { children: ReactNode }) => {
+  return <SessionProvider>{children}</SessionProvider>
```

Comment:
```suggestion
export const AuthedSession = ({ children }: { children: ReactNode }) => {
const session = useSession()

  // sign user out if session is expired
  useEffect(() => {
    if (
      !session ||
      !session.data ||
      new Date(session.data.expires) < new Date()
    ) {
      signOut()
    }
  }, [session])
  
  return <>{children}</>
}

  return (<SessionProvider>
  <AuthedSession >
  {children}
  </AuthedSession>
  </SessionProvider>)
```

Apologies for messy code but something like this